Model/Method/AbstractMethod.php

```diff
@@ -2430,5 +2432,38 @@ public function getShippingAddress($payment, &$isAbsent = false)
         }
         return $shippingAddress;
     }
+
+    public function checkTotalGrossAmount($requestData, $payment)
+    {
+        $order = $payment->getOrder();
+        $itemsTotalAmount = 0;
+        $count = 1;
+        $requestData2 = [];
+        foreach ($requestData as $item){
+            if(isset($item['GroupID']) && $item['GroupID']>0){
+                if($item['Name'] == 'Quantity'){
+                    $requestData2[$item['GroupID']]['Quantity'] = $item['_'];
+                }
+                if($item['Name'] == $this->getPriceFieldName()){
+                    $requestData2[$item['GroupID']][$this->getPriceFieldName()] = $item['_'];
+                }
+            }
+        }
+
+        foreach ($requestData2 as $key=>$item){
```

Comment:
could you please adjust spaces indentation in this new function

Reply:
sure, fixed, thank you